src/asset-management/AaveManager.sol

```diff
@@ -51,18 +51,18 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
     //////////////////////////////////////////////////////////////////////////*/
 
     /// @dev returns the balance of the token managed by various markets in the native precision
-    function getBalance(IAssetManagedPair aOwner, address aToken) external view returns (uint112 rTokenBalance) {
+    function getBalance(IAssetManagedPair aOwner, address aToken) external view returns (uint104 rTokenBalance) {
         return _getBalance(aOwner, aToken);
     }
 
-    function _getBalance(IAssetManagedPair aOwner, address aToken) private view returns (uint112 rTokenBalance) {
+    function _getBalance(IAssetManagedPair aOwner, address aToken) private view returns (uint104 rTokenBalance) {
         address lAaveToken = _getATokenAddress(aToken);
         uint256 lTotalShares = totalShares[lAaveToken];
         if (lTotalShares == 0) {
             return 0;
         }
         rTokenBalance =
-            uint112(shares[aOwner][aToken] * IERC20(lAaveToken).balanceOf(address(this)) / totalShares[lAaveToken]);
+            uint104(shares[aOwner][aToken] * IERC20(lAaveToken).balanceOf(address(this)) / totalShares[lAaveToken]);
```

Comment:
Overflow?

Reply:
I think we should be good cuz `balanceOf` returns a uint256 which the `uint104` will be implicitly cast into during the operation

Reply:
There is no overflow checking when you cast uint256 to uint104 in solidity pretty sure?

Reply:
that's correct. No overflow check. But with the amounts we are managing, it's unlikely that we'd reach that limit. However, if we're still worried, we can use openzeppelin's `SafeCast::toUint104()` to ensure that it is indeed within range

Reply:
Yeah, I'd probably prefer a safeCast here even though its extremely unlikely its still possible

Reply:
okay created #140 to make sure we don't miss it. 

Resolving for now